### PR TITLE
chore(local): Add weave as a service in k8s

### PR DESCRIPTION
**Pull request by yogeshg**

### CODE REVIEW

**Review by vanpelt — APPROVED** [review]
🔥

File: main.tf [hunk 1]
```diff
@@ -146,6 +147,35 @@ resource "kubernetes_deployment" "wandb" {
...
+              field_ref {
+                field_path = "status.hostIP"
+              }
+            }
+          }
+
+          env {
+            name = "GORILLA_STATSD_PORT"
+            value = "8125"
+          }
```
Comment (vanpelt) [inline main.tf:167]: I think these need to be opt-in.  Lot's of clusters won't have a DD agent accepting StatsD metrics.

File: weave.tf [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+          }
+
+          env {
+            name = "DD_AGENT_HOST"
+            value_from {
+              field_ref {
+                field_path = "status.hostIP"
+              }
+            }
+          }
```
Comment (vanpelt) [inline weave.tf:76]: I'm a little worried weave might try to talk to an agent even if `WEAVE_ENABLE_DATADOG` is set to false.  Ideally we would only specify these env vars if we know we have an agent running.

**Review by vanpelt — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (yogeshg) [reply main.tf]: hmm, lemme just remove these two then, it can be passed via other_wandb_env

File: weave.tf (hunk 2 above)
Comment (yogeshg) [reply weave.tf:70]: good call. added a dynamic block.

**Review by vanpelt — APPROVED** [review]

**Comment by jsbroks** [comment]
This PR is included in version 1.8.0 :tada:

**Pull request merged**